src/js/pre-logic.js

```diff
@@ -135,39 +137,50 @@ function checkWall(state) {
   return (spriteState) => {
     return state.board.walls.find((element) => {
       return element.x === spriteState.x && element.y === spriteState.y;
-    });
+    }) !== undefined;
   };
 }
 
 const checkIfWall = checkWall(currentState);
 
 function crunchState(state, action) {
-  let buildState;
-  const pacManInput = action.input === 'nope' ? state.direction : action.input;
-  let newPacmanState = crunchSpriteState(state.pacman, pacManInput);
-  let isAWall = checkIfWall(newPacmanState);
-  if (isAWall) {
-    buildState = state;
-  } else {
-    buildState = {...state, pacman: newPacmanState};
-  }
-  let newGhostState = crunchSpriteState(state.red, state.red.direction);
-  isAWall = checkIfWall(newGhostState);
-  if (isAWall) {
-    let aNewDirection = pickRanDir();
-    buildState = {...buildState, red: {...buildState.red, direction: aNewDirection}};
-  } else {
-    buildState = {...buildState, red: newGhostState};
-  }
-  newGhostState = crunchSpriteState(state.orange, state.orange.direction);
-  isAWall = checkIfWall(newGhostState);
-  if (isAWall) {
-    let aNewDirection = pickRanDir();
-    buildState = {...buildState, orange: {...buildState.orange, direction: aNewDirection}};
+  if (!state.collision) {
+    let buildState;
+    const pacManInput = action.input === 'nope' ? state.direction : action.input;
+    let newPacmanState = crunchSpriteState(state.pacman, pacManInput);
+    let isAWall = checkIfWall(newPacmanState);
+    if (isAWall) {
+      buildState = state;
+    } else {
+      buildState = {...state, pacman: newPacmanState};
+    }
+    let newGhostState = crunchSpriteState(state.red, state.red.direction);
+    isAWall = checkIfWall(newGhostState);
+    if (isAWall) {
+      let aNewDirection = pickRanDir();
+      buildState = {...buildState, red: {...buildState.red, direction: aNewDirection}};
+    } else {
+      buildState = {...buildState, red: newGhostState};
+    }
+    newGhostState = crunchSpriteState(state.orange, state.orange.direction);
+    isAWall = checkIfWall(newGhostState);
```

Comment:
This code is repeated three times. One for pac-man, one for red and one for orange. What if at the beginning og the function you place them in `[state.pacman, state.orange, state.red]`? Would you be able to make your code more concise?